src/pages/home/ReportScreen.js

```diff
@@ -276,111 +276,107 @@ class ReportScreen extends React.Component {
                                             isComposerFullSize={this.props.isComposerFullSize}
-                                            onSubmitComment={this.onSubmitComment}
-                                            policies={this.props.policies}
+                                            policy={policy}
                                         />
-                                    </>
-                                )}
+                                    )}
 
-                                {!this.isReportReadyForDisplay() && (
-                                    <ReportFooter
-                                        shouldDisableCompose
-                                        isOffline={this.props.network.isOffline}
-                                    />
-                                )}
-                            </View>
-                        </DragAndDropProvider>
-                    </FullPageNotFoundView>
-                </ScreenWrapper>
-            </ReportScreenContext.Provider>
+                                    {/* Note: The report should be allowed to mount even if the initial report actions are not loaded. If we prevent rendering the report while they are loading then
+                            we'll unnecessarily unmount the ReportActionsView which will clear the new marker lines initial state. */}
+                                    {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions || isLoading) && (
+                                        <ReportActionsSkeletonView containerHeight={this.state.skeletonViewContainerHeight} />
+                                    )}
+
+                                    {this.isReportReadyForDisplay() && (
+                                        <>
+                                            <ReportFooter
+                                                pendingAction={addWorkspaceRoomOrChatPendingAction}
+                                                isOffline={this.props.network.isOffline}
+                                                reportActions={this.props.reportActions}
+                                                report={this.props.report}
+                                                isComposerFullSize={this.props.isComposerFullSize}
+                                                onSubmitComment={this.onSubmitComment}
+                                                policies={this.props.policies}
+                                            />
+                                        </>
+                                    )}
+
+                                    {!this.isReportReadyForDisplay() && (
+                                        <ReportFooter
+                                            shouldDisableCompose
+                                            isOffline={this.props.network.isOffline}
+                                        />
+                                    )}
```

Comment:
These should be a ternary

Reply:
All these are part of linting (the component it is now wrapped into a new context provider), I have not changed or touched the internal logic. I can do it if you want, but it falls out of the scope of the performance improvements